java/compaction/compaction-core/src/main/java/sleeper/compaction/job/commit/CompactionJobCommitterOrSendToLambda.java

```diff
@@ -55,6 +67,10 @@ public void commit(CompactionJob job, CompactionJobFinishedEvent.Builder finishe
         } else {
             LOGGER.info("Committing compaction job {} inside compaction task", job.getId());
             CompactionJobCommitter.updateStateStoreSuccess(job, finishedEvent.getSummary().getRecordsWritten(), stateStoreProvider.getByTableId(job.getTableId()));
+            statusStore.jobCommitted(CompactionJobCommittedEvent.compactionJobCommitted(job, timeSupplier.get())
+                    .jobRunId(finishedEvent.getJobRunId())
+                    .taskId(finishedEvent.getTaskId())
+                    .build());
             statusStore.jobFinished(finishedEvent);
```

Comment:
I think the finished event should be reported as soon as the compaction has finished, whether it's synchronous or asynchronous.